File: ui/articles-aem-renderer/src/main/java/org/godtools/uiarticles_aem_renderer/db/ArticleRepository.java
package org.godtools.uiarticles_aem_renderer.db;

import android.app.Application;
import android.arch.lifecycle.LiveData;

import org.godtools.uiarticles_aem_renderer.model.Article;

import java.util.List;

/**
 *
 */
public class ArticleRepository {

    private ArticleDao mArticleDao;
    private List<Article> mAllArticles;

    public ArticleRepository(Application _application) {
        ArticleRoomDatabase db = ArticleRoomDatabase.getINSTANCE(_application);
        mArticleDao = db.mArticleDao();
        mAllArticles = mArticleDao.getAllArticles();
    }


    /**
     *
     * @return
     */
    List<Article> getmAllArticles(){
        return mAllArticles;
    }
}

File: ui/articles-aem-renderer/src/main/java/org/cru/godtools/articles/aem/db/ArticleRepository.kt
package org.cru.godtools.articles.aem.db

import android.arch.persistence.room.Dao
import android.arch.persistence.room.Transaction
import org.cru.godtools.articles.aem.model.Article

@Dao
abstract class ArticleRepository internal constructor(private val db: ArticleRoomDatabase) {
    @Transaction
    open fun updateContent(article: Article) {
        db.articleDao().updateContent(article.uri, article.contentUuid, article.content)
        db.resourceRepository().storeResources(article, article.mResources)
    }
}
